master/internal/api_user.go

```diff
@@ -50,6 +50,10 @@ func clearUsername(targetUser model.User, name string, minLength int) (*string,
 	return &clearName, nil
 }
 
+func checkEmptyProtoAgentUserGroup(aug *userv1.AgentUserGroup) bool {
+	return aug.AgentUid == nil && aug.AgentGid == nil && aug.AgentUser == nil && aug.AgentGroup == nil
+}
+
 func validateProtoAgentUserGroup(aug *userv1.AgentUserGroup) error {
 	if aug.AgentUid == nil || aug.AgentGid == nil || aug.AgentUser == nil || aug.AgentGroup == nil {
```

Comment:
it seems this check is not valuable any more because we now totally allow empty ones. maybe instead of this checkEmptyProtoAgentUserGroup + validateProtoAgentUserGroup logic it'd be simpler to have a method which converts `userv1.AgentUserGroup -> model.AgentUserGroup` supporting either empty or non-empty groups.

it could probably be in `model/agent_user_group.go`, something like
```
func AgentUserGroupFromProto(aug *userv1.AgentUserGroup) (model.AgentUserGroup, error)
```

that should simplify the code in `api_user`.

Reply:
What do you mean by empty groups? 
Is it like the agent user group validation in workspace, where AgentUid and AgentUser can both be non-empty or empty, also AgentGid and AgentGroup can both be non-empty or empty?

Reply:
sorry, by "empty group" I mean this special proto agentUserGroup being '{}' input case.

with your PR:
- `validateProtoAgentUserGroup` doesn't allow {} (like before)
- and we have special code outside of `validate...` to enable it

I think `validateProtoAgentUserGroup` is now meaningless and it can be simplified by having single `AgentUserGroupFromProto` proto parser

Reply:
Got it, thanks!